src/tfs/tfs.c

```diff
@@ -427,7 +427,11 @@ static fs_status create_extent(filesystem fs, range blocks, boolean uninited, ex
     u64 nblocks = MAX(range_span(blocks), MIN_EXTENT_SIZE >> fs->blocksize_order);
 
     tfs_debug("create_extent: blocks %R, uninited %d, nblocks %ld\n", blocks, uninited, nblocks);
-
+    if (!filesystem_reserve_log_space(fs, &fs->next_extend_log_offset, 0, 0) ||
```

Comment:
Since the log flush runs behind file writes, it is possible for a log extend or new new log to grab the reserved log space but be unable to allocate a new log space at that point. Without this new log space in reserve, we can't let new extents be created since they would be at risk of not being recorded if the current log space runs out. This makes sure there's always log space before doing any writes, and also it fills the role of allocating log space after freeing space from an exhaustion event as opposed to checking after each extent is destroyed. Hopefully this makes sense.

Reply:
Yes, this makes sense to me now. I didn't realize that it was meant to recover from a condition where one of the extension allocs failed at an earlier point.